engine/access/ingestion/engine.go

```diff
@@ -33,12 +33,18 @@ const collectionCatchupDBPollInterval = 10 * time.Millisecond
 const fullBlockUpdateInterval = 1 * time.Minute
 
 // a threshold of number of blocks with missing collections beyond which collections should be re-requested
+// this is to prevent spamming the collection nodes with request
 const missingCollsForBlkThreshold = 100
 
+// a threshold of block height beyond which collections should be re-requested (regardless of the number of blocks for which collection are missing)
+// this is to ensure that if a collection is missing for a long time (in terms of block height) it is eventually re-requested
+const missingCollsForHeightThreshold = 100
+
 var defaultCollectionCatchupTimeout = collectionCatchupTimeout
 var defaultCollectionCatchupDBPollInterval = collectionCatchupDBPollInterval
 var defaultFullBlockUpdateInterval = fullBlockUpdateInterval
 var defaultMissingCollsForBlkThreshold = missingCollsForBlkThreshold
+var defaultMissingCollsForHeightThreshold = missingCollsForHeightThreshold
```

Comment:
These seem like they should be config fields on the `Engine` instance rather than global variables

Reply:
so these private fields were meant to help white box testing where they can be modified as part of the unit test. Yeah, I could add it to the engine too and make them configurable (but may do that as a different issue).